lib/pages/preferences_page.dart

```diff
@@ -10,102 +11,80 @@ class PreferencesPage extends StatefulWidget {
 }
 
 class _PreferencesPageState extends State<PreferencesPage> {
-  Future<SharedPreferences> _preferences;
 
   @override
   void initState() {
     super.initState();
-    _preferences = SharedPreferences.getInstance();
   }
 
   @override
   Widget build(BuildContext context) {
-    return ThemeSwitchingArea(
-      child: Scaffold(
-        backgroundColor: Theme.of(context).primaryColor,
-        appBar: AppBar(title: Text("Préférences")),
-        body: _buildPreferencesContent(),
-      ),
+    return Scaffold(
+      backgroundColor: Theme.of(context).primaryColor,
+      appBar: AppBar(title: Text("Préférences")),
+      body: _buildPreferencesContent(),
     );
   }
 
   Widget _buildPreferencesContent() {
-    return FutureBuilder<SharedPreferences>(
-      future: _preferences,
-      builder: (context, snapshot) {
-        switch (snapshot.connectionState) {
-          case ConnectionState.waiting:
-            return Center(
-              child: Column(
+    return BlocBuilder<PreferencesBloc, PreferencesState>(
+      bloc: BlocProvider.of<PreferencesBloc>(context),
+      builder: (context, state) {
+        if (state is PreferencesLoadedState) {
+          return ListView.builder(
```

Comment:
For small list, you can just use `ListView` (without `builder`) and set its children which is easier and has no performance issue (but keep `ListView.builder` for long/dynamic list 😄 )